lib/salus/scanners/base.rb

```diff
@@ -76,11 +79,26 @@ def run!(salus_report:, required:, pass_on_raise:, reraise:)
       salus_report.add_scan_report(@report, required: required)
 
       begin
-        @report.record { run }
+        @report.record do
+          Timeout.timeout(scanner_timeout) { run }
+        end
 
         if @report.errors.any?
           pass_on_raise ? @report.pass : @report.fail
         end
+      rescue Timeout::Error
+        error_message = "Scanner #{name} timed out during execution"
+        timeout_error_data = {
+          message: error_message,
+          error_class: ScannerTimeoutError,
+          backtrace: []
```

Comment:
I think we should delete `backtrace: []`, because it's always a hardcoded empty array.